### PR TITLE
Fix assert filename/line number when testing 

### PR DESCRIPTION
**Pull request by FergusonAJ**
Looking into #445, I noticed a comment in `tests/base/assert.cpp`:
```
/* not sure why file is assert.h and line_num is 95, is this expected behavior?
std::string filen = emp::assert_fail_info.filename;
std::string test_filen = "assert.h";
REQUIRE(filen.substr(filen.size()-test_filen.size()) == test_filen);
REQUIRE(emp::assert_fail_info.line_num == 95);
*/
```

There was a simple fix! The previous code in `include/emp/base/_tdebug_assert_trigger.hpp` was as follows: 
```
template <typename... EXTRA>
bool assert_trigger(std::string filename, size_t line, std::string expr, EXTRA &&...) {
    emp::assert_fail_info.filename = __FILE__;
    emp::assert_fail_info.line_num = __LINE__;
    ... // rest look good   
}
```

Note the body of the function uses `__FILE__` and `__LINE__`, not the filename and line number that were passed in! Thus the filename and line number were always set to `_tdebug_assert_trigger.hpp` and line in that file, respectively.  
Swapping those out for the actual parameters did the trick! 

Otherwise, I updated the test file (`tests/base/assert.h`), and also streamlined it a little.

### CODE REVIEW

File: tests/base/assert.cpp [hunk 1]
```diff
@@ -31,44 +31,41 @@ TEST_CASE("Test assert", "[base]")
...
-  EMP_TEST_VALUE(emp::assert_last_fail, "1");
-  EMP_TEST_VALUE(emp::assert_fail_info.filename, "assert.cc");
-  EMP_TEST_VALUE(emp::assert_fail_info.line_num, "31");
-  EMP_TEST_VALUE(emp::assert_fail_info.error, "false");
-*/
-  // if (emp::assert_fail_info.filename != "assert.cc") std::cerr << "Failed case 6!" << std::endl;
-  // if (emp::assert_fail_info.line_num != __LINE__ - 3) std::cerr << "Failed case 7!" << std::endl;
-  // if (emp::assert_fail_info.error != "false") std::cerr << "Failed case 8!" << std::endl;
+  expected_line_num = __LINE__ + 1; // This line must precede the assert for line num to pass
+  emp_assert(2 > 3); // False! Trigger the assert
```
Comment (mmore500) [inline tests/base/assert.cpp:61]: Check indentation

File: tests/base/assert.cpp [hunk 2]
```diff
@@ -31,44 +31,41 @@ TEST_CASE("Test assert", "[base]")
 
 	// Best way to test different behavior of emp_assert based on compiler flags?
 
+  int expected_line_num = __LINE__ + 1; // This line must precede the assert for line num to pass
```
Comment (mmore500) [inline tests/base/assert.cpp:34]: Const? Check indentation too

Is this file mixed tabs and spaces????

File: tests/base/assert.cpp [hunk 3]
```diff
@@ -31,44 +31,41 @@ TEST_CASE("Test assert", "[base]")
...
 	emp_assert(false);
 	REQUIRE(emp::assert_last_fail);
 
-	/* not sure why file is assert.h and line_num is 95, is this expected behavior?
+  // Test filename and line number
+  // Trim the filename in case it runs from another directory
 	std::string filen = emp::assert_fail_info.filename;
```
Comment (mmore500) [inline tests/base/assert.cpp:40]: Const and we should just use the full word filename — we’re not paying by the character

File: tests/base/assert.cpp [hunk 4]
```diff
@@ -31,44 +31,41 @@ TEST_CASE("Test assert", "[base]")
...
-	std::string test_filen = "assert.h";
+	std::string test_filen = "assert.cpp";
 	REQUIRE(filen.substr(filen.size()-test_filen.size()) == test_filen);
```
Comment (mmore500) [inline tests/base/assert.cpp:42]: Use Catch matcher? https://github.com/catchorg/Catch2/blob/devel/docs/matchers.md#using-matchers

**Review by mmore500 — APPROVED** [review]
Good catch on a subtle bug. A few mostly formatting comments.

**Comment by FergusonAJ** [comment]
Implemented Matthew's suggestions (great catch on the indentation!) ~Merging!

**Pull request merged**